### DISCUSSION TITLE
An updated example for Next.js 13 (app router)

### DISCUSSION THREAD

**Discussion by utkarshdubey** [family/connectkit#316] [Q&A] [answered]
I was wondering if anyone's been using `connectkit` in a Next.js 13 project using the app router. I've been having many problems trying to figure it out and haven't had any luck so far.

**Comment by lochie** [comment, marked as answer]
You can find an example of this here:
https://github.com/m1guelpf/nextjs13-connectkit-siwe

**Reply by utkarshdubey** [reply]
The example doesn't cover fetching session server side to add gating to specific pages.

**Reply by alex-grover** [reply]
@utkarshdubey here's a project of mine with an implementation of SIWE based on the above. You can add gating by getting a session from the cookies.

https://github.com/alex-grover/later.wtf/blob/main/app/api/casts/route.ts#L22-L23

**Reply by utkarshdubey** [reply]
Is there a way I could do this in a page.tsx file since I need the user's wallet address to make a db query.

**Reply by alex-grover** [reply]
yep, you can get cookies in a page component like this:

```tsx
import { cookies } from 'next/headers'

export default function YourPage() {
  const { address } = await Session.fromCookies(cookies())
  // ...
}
